docs/guides/react-server-cli.md

```diff
@@ -1,113 +1,198 @@
+
+By default this is created at the site's root directory as `routes.json`.
 
-There are three ways to pass options to the CLI, through the command line, `.reactserverrc` JSON files, or as a `reactServer` entry in `package.json` files. It searches for options this way:
+```json
+{
+	"middleware": [
+		"middleware/FooMiddleware",
+		"middleware/BarMiddleware"
+	],
+	"routes": {
+		"BazPage": {
+			"path": ["/"],
+			"page": "pages/BazPage"
+		},
+		"BakPage": {
+			"path": ["/bak"],
+			"page": "pages/BakPage"
+		}
+	}
+}
+```
 
-1. If there are any options arguments on the command line, they are used. For the options which aren't specified:
-1. `react-server-cli` looks at the current directory.
- 1. If there is a JSON file named `.reactserverrc` in the directory, its settings are used and we skip to step #4. Otherwise:
- 1. If there is a `package.json` file in the current directory with an entry named `reactServer`, its settings are used and we skip to step #4. Otherwise:
-1. Go back to step 2 in the parent of the current directory. Repeat until you either find a config or hit the root directory.
-1. If there are any options that still aren't specified, the defaults are used.
+## Server config
 
-Note that the programmatic API also searches for config files, although options sent in explicitly to the API function override the config file.
+You can define JSON options either in a `.reactserverrc` or in a
+`reactServer` object in your `package.json`.
 
-### JSON options can be set per environment
+You can provide environment-specific values in a sub-object at the key `env`.
 
-If you are using either `.reactserverrc` or `package.json` to set your react-server options, you can provide environment-specific values in a sub-object at the key `env`. It looks like this:
+It looks like this:
 
 ```json
 {
+	"routes": "routes.json",
 	"port": "5000",
 	"env": {
-		"staging": {
-			"port": "4000"
-		},
 		"production": {
 			"port": "80"
 		}
 	}
 }
 ```
 
-The values in a particular environment override the main settings. In this example configuration `port` will be set to 80 if `process.env.NODE_ENV` is `production`, 4000 if `process.env.NODE_ENV` is `staging`, and 5000 for any other situation.
+The values in a particular environment override the main settings. In this
+example configuration `port` will be set to 80 if `process.env.NODE_ENV` is
+`production`, and 5000 otherwise.
+
+
+## Options
+Smart defaults are the goal, and `react-server-cli` has two base modes:
+**development** and **production**. `react-server-cli` will determine which
+base mode it's in by looking at the NODE_ENV environment variable. If it's not
+"production", then `react-server-cli` will assume we are in development mode.
+
+### Ways to add options
+
+There are three ways to pass options to the CLI, through the command line,
+`.reactserverrc` JSON files, or as a `reactServer` entry in `package.json`
+files. If there's no config file (or package.json config) in the current
```

Comment:
I don't actually know.  I had totally forgotten about the `package.json` configuration until I started editing this doc. 😁